Extended electroNP flowsheet with ASM and AD

luohezhiming opened the pull request:
## Fixes/Resolves:

(replace this with the issue # fixed or resolved, if no issue exists then a brief statement of what this PR does)

## Summary/Motivation:


## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

luohezhiming committed:
add

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

luohezhiming committed:
delete redundant files

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

Commit:
dewtering unit model

agarciadiego committed:
Merge branch 'watertap-org:main' into dewatering

Commit:
removing imports

Commit:
black

Commit:
black and doc error

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

Commit:
testing scaling

Commit:
pylint errors 1

Commit:
change blk to self

Commit:
pylint errors 2

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

Commit:
adding test and list error

Commit:
thickener unit model

Commit:
unit consistency check

agarciadiego committed:
Merge branch 'main' into dewatering

Commit:
applying comments

Commit:
fix typo

Commit:
fix typo 2

agarciadiego committed:
Update watertap/unit_models/thickener.py

Co-authored-by: Adam Atia <aatia@keylogic.com>

agarciadiego committed:
Update watertap/unit_models/dewatering.py

Co-authored-by: Adam Atia <aatia@keylogic.com>

Commit:
chaine unit name in tests

Commit:
changing class name

adam-a-a committed:
cleanup unused imports and add time index to TSS in dewaterer

adam-a-a committed:
remove unused imports from test_dewatering

adam-a-a committed:
more cleanup

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

Commit:
BSM2 full flowsheet

Commit:
gettting branch up-to-date

Commit:
rewriting flowsheet

Commit:
commenting translator error

Commit:
import and prop cleanup

Commit:
fixing failed import

agarciadiego committed:
fiing import

agarciadiego committed:
Merge branch 'dewatering' of https://github.com/agarciadiego/watertap into dewatering

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

agarciadiego committed:
fixing expresions that give values below 0

agarciadiego committed:
Merge branch 'main' of https://github.com/watertap-org/watertap into dewatering

agarciadiego committed:
Merge branch 'watertap-org:main' into dewatering

agarciadiego committed:
Merge branch 'dewatering' of https://github.com/agarciadiego/watertap into dewatering

agarciadiego committed:
changing to smooth max

agarciadiego committed:
fixing smooth max

agarciadiego committed:
fixing units error

agarciadiego committed:
uncommenting einitialization error

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

luohezhiming committed:
add sensitivity analysis

luohezhiming committed:
add sweep analysis

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

luohezhiming committed:
Merge branch 'main' into electroNP_flowsheet_analysis

luohezhiming committed:
add sweep analysis

luohezhiming committed:
add more analysis

luohezhiming committed:
add analysis

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

luohezhiming committed:
revise file

luohezhiming committed:
add more analysis

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

luohezhiming committed:
Merge branch 'main' into electroNP_flowsheet_analysis

luohezhiming committed:
revise flowsheet

MarcusHolly committed:
Add i_NOx_N2 Var to modified_asm2d_reactions

luohezhiming committed:
revise analysis

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

MarcusHolly committed:
minor updates

MarcusHolly committed:
finish 1st implementation of translator block

MarcusHolly committed:
Make corrections to implementation

MarcusHolly committed:
Add test file

MarcusHolly committed:
Merge branch 'main' into asm2d/adm1

Commit:
Merge branch 'upmain' into issue934dewatering

# Conflicts:
#	watertap/unit_models/translators/translator_asm1_adm1.py

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

luohezhiming committed:
Merge branch 'main' into electroNP_flowsheet3

MarcusHolly committed:
Rework expressions with no indices

MarcusHolly committed:
Finish base implementation - need to clean up

Commit:
Correct unit convertion, bar to Pa

MarcusHolly committed:
Add some initial testing

MarcusHolly committed:
Reorder file so that DecaySwitch.on and off are completely separate rather than interwoven

MarcusHolly committed:
Replaces 1e-9s with eps

luohezhiming committed:
add sweep analysis

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

luohezhiming committed:
Merge branch 'main' into electroNP_flowsheet3

luohezhiming committed:
delete scaling factors

luohezhiming committed:
revise test file

MarcusHolly committed:
Fix DOF error

luohezhiming committed:
revise opex frac

luohezhiming committed:
plinting code

MarcusHolly committed:
Clean up code

MarcusHolly committed:
Merge branch 'main' into asm2d/adm1

MarcusHolly committed:
Fix DecaySwitch error

luohezhiming committed:
Update watertap/costing/units/electroNP.py

Co-authored-by: Adam Atia <aatia@keylogic.com>

luohezhiming committed:
Update watertap/costing/units/electroNP.py

Co-authored-by: Adam Atia <aatia@keylogic.com>

luohezhiming committed:
Update watertap/costing/units/electroNP.py

Co-authored-by: Adam Atia <aatia@keylogic.com>

luohezhiming committed:
Update watertap/examples/flowsheets/case_studies/electroNP/multi_sweep.py

Co-authored-by: Adam Atia <aatia@keylogic.com>

luohezhiming committed:
clean up the code

luohezhiming committed:
revise value

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

luohezhiming committed:
Merge branch 'main' into electroNP_flowsheet3

MarcusHolly committed:
Add documentation for ASM2d/ADM1

MarcusHolly committed:
Fix doc formatting

MarcusHolly committed:
Fix documentation formatting again

MarcusHolly committed:
Remove step labels from tables

MarcusHolly committed:
Potential documentation fix

MarcusHolly committed:
Documentation fix

MarcusHolly committed:
Remove step labels

MarcusHolly committed:
Doc fix

MarcusHolly committed:
Update index

MarcusHolly committed:
Merge branch 'main' into asm2d/adm1

luohezhiming committed:
add test for display_costing

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

luohezhiming committed:
Merge branch 'main' into electroNP_flowsheet3

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

adam-a-a committed:
Merge branch 'main' into asm2d/adm1

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

luohezhiming committed:
Merge branch 'main' into electroNP_flowsheet3

Commit:
Revised AD model that reproduce results from the paper

Commit:
Add domains to avoid evaluation of log(negative)

Commit:
Update ADM1 flowsheet test to be consistent with the paper

MarcusHolly committed:
Address merge conflicts

luohezhiming committed:
add dewatering unit

adam-a-a committed:
Merge branch 'main' into asm2d/adm1

MarcusHolly committed:
Convert SIC_output to kg/m3 using pyunits.convert

MarcusHolly committed:
merge main

MarcusHolly committed:
Replace all instances of g/m3 with kg/m3, so no longer need 1000s in unit model

MarcusHolly committed:
Merge branch 'main' into asm2d/adm1

MarcusHolly committed:
minor clean up

MarcusHolly committed:
Add validation tests

MarcusHolly committed:
Unmute remaining tests

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

luohezhiming committed:
Merge branch 'asm2d/adm1' into electroNP_flowsheet4

Commit:
Update temprature bounds and intial values

Commit:
Merge branch 'upmain' into issue934dewatering

# Conflicts:
#	watertap/unit_models/dewatering.py
#	watertap/unit_models/thickener.py

luohezhiming committed:
add extended flowsheet

luohezhiming committed:
ASM2D

Commit:
BSM2 flowsheet pass with uniform temperature

agarciadiego committed:
merging conlfict

Commit:
Modify models to use default solver options in BSM2 flowsheet, pass

Commit:
Update tests

agarciadiego committed:
Merge branch 'dewatering' into issue934dewatering

agarciadiego committed:
Merge pull request #1 from lxhowl/issue934dewatering

Issue934dewatering

agarciadiego committed:
changing unit model to separator

agarciadiego committed:
updating flowsheet

agarciadiego committed:
Merge branch 'watertap-org:main' into electronP_separator

agarciadiego committed:
running black

agarciadiego committed:
Merge branch 'electronP_separator' of https://github.com/agarciadiego/watertap into electronP_separator

agarciadiego committed:
running pylint

Commit:
Update ASM model variations and translators

Commit:
Update ADM model variation and tests

agarciadiego committed:
Merge branch 'main' of https://github.com/watertap-org/watertap into dewatering

agarciadiego committed:
deleteing old files and rearranging

agarciadiego committed:
running black

agarciadiego committed:
deleting unused imports

agarciadiego committed:
thermo vapor test

MarcusHolly committed:
Merge branch 'main' into asm2d/adm1

MarcusHolly committed:
Update influent flow_vol for test file

Commit:
Merge commit 'c29e058a81ac179e2c2e90e41d0988b843e4bc7c' into issue934dewatering

Commit:
Merge commit '277b3cea57e9981a3838852a88a93d75183d1290' into issue934dewatering

Commit:
Merge commit 'acff3ef4871dd2738889d97c179d6054aad064ec' into issue934dewatering

Commit:
Run black and update electroNP flowsheet used AD

agarciadiego committed:
Merge pull request #3 from lxhowl/issue934dewatering

Update AD & AS models and tests

agarciadiego committed:
skip mac test

luohezhiming committed:
add extended flowsheet

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

MarcusHolly committed:
Fix S_IC initialization issue

MarcusHolly committed:
Move i_PSI param from prop model to translator unit model

MarcusHolly committed:
Merge branch 'main' into asm2d/adm1

MarcusHolly committed:
Remove unused imports

MarcusHolly committed:
Merge branch 'main' into asm2d/adm1

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

luohezhiming committed:
Merge branch 'asm2d/adm1' into electroNP_flowsheet4

luohezhiming committed:
Merge branch 'dewatering' into electroNP_flowsheet4

luohezhiming committed:
revise file

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

luohezhiming committed:
Merge branch 'main' into electroNP_flowsheet4

agarciadiego committed:
changing scaling

agarciadiego committed:
running black

luohezhiming committed:
Modify model

agarciadiego committed:
solving pylint

agarciadiego committed:
scaling

adam-a-a commented:
@luohezhiming - Just for the record, this includes @Xinhong's revisions on modified ASM2d, correct? If so, that would allow us to remove those changes on the BSM2 flowsheet PR (#1038)

luohezhiming commented:
> @luohezhiming - Just for the record, this includes @XinHong's revisions on modified ASM2d, correct? If so, that would allow us to remove those changes on the BSM2 flowsheet PR (#1038)

Yeah, this included @XinHong's reformulation for ASM2D and ADM1 model.

luohezhiming commented:
@XinHong, this PR included your reformulations of the modified ASM2D and modified ADM1 models. A new `flowsheet modified_BSM2_flowsheet.py` is added in this PR, it is a sub-flowsheet of BSM2 without recycles but using modified ASM2D and modified ADM1 properties models instead of ASM1 and ADM1. Your reformulations really help to resolve scaling issues and the flowsheet can be solved to optimal, but I get an error `Cannot recompute multipliers for feasibility problem.  Error in eq_mult_calculator` when I check the solver log. I believe there are still some scaling issues or degeneracy, if you could help to check it, we will really appreciate you!

agarciadiego committed:
revert to old names

luohezhiming committed:
resolve scaling

luohezhiming committed:
Merge branch 'electronP_separator' into electroNP_flowsheet4

luohezhiming committed:
add costing and merge new electroNP_ZO

luohezhiming committed:
adjust scaling

luohezhiming committed:
revise flowsheet

luohezhiming committed:
flowsheet with electroNP_v2

luohezhiming committed:
electroNP_v2_with_no_costing

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

luohezhiming committed:
Merge branch 'main' into electroNP_flowsheet4

luohezhiming committed:
working v2 version for test

luohezhiming committed:
extended electroNP flowsheet v2

luohezhiming committed:
Merge branch 'main' of https://github.com/watertap-org/watertap

luohezhiming committed:
Merge branch 'main' into electroNP_flowsheet4

luohezhiming committed:
Merge branch 'main' into electroNP_flowsheet4

luohezhiming committed:
Merge branch 'electroNP_flowsheet4' of https://github.com/luohezhiming/watertap into electroNP_flowsheet4

adam-a-a commented on watertap/examples/flowsheets/case_studies/BSM2/BSM2_flowsheet2.py:
```diff
@@ -0,0 +1,422 @@
+#################################################################################
```
@luohezhiming I am not sure where this is coming from. I don't think it is from main because I reorganized this and renamed this BSM2 flowsheet, so you should start by making sure you're using the most up-to-date work

luohezhiming commented:
This PR is replaced by PR #1112
